pkgs/applications/editors/tmcbeans/default.nix

```diff
@@ -0,0 +1,82 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, makeDesktopItem
+, which
+, unzip
+, libicns
+, imagemagick
+, jdk
+, copyDesktopItems
+}:
+
+let
+  version = "1.4.0";
+  desktopItem = makeDesktopItem {
+    name = "tmcbeans";
+    exec = "tmcbeans";
+    comment = "Integrated Development Environment";
+    desktopName = "Apache NetBeans IDE with TMC";
+    genericName = "Integrated Development Environment";
+    categories = [ "Development" ];
+    icon = "tmcbeans";
+  };
+in
+stdenv.mkDerivation {
+  pname = "tmcbeans";
+  inherit version;
+  src = fetchurl {
+    url = "https://download.mooc.fi/tmcbeans/installers/tmcbeans_${version}.zip";
+    hash = "sha512-kugZ7DAarbKK8RvOx7VYIpCwNhm3QVBlvqy+R7oeDDTT0LeDtHsWNIpIQSgT7t9MExfG1X9bqCqspJwYQxUG5g==";
+  };
+
+  patchPhase = ''
+    # Perform some path patching.
+    patchShebangs .
```

Comment:
IIRC, this is done automatically, at least for stuff in `/bin`. I could be wrong, though. Could you please recheck that? 

A nitpick: it's better to have this in `postFixup` since it fixes the actual output and not a build script.